### PR TITLE
VoyageAI vectorizer and reranker

### PR DESCRIPTION
**Pull request by fzowl**

### CODE REVIEW

**Review by bsbodden — COMMENTED** [review]
@tylerhutcherson LGTM!

**Comment by bsbodden** [comment]
@fzowl there are tests failing in our test matrix, can you address them before the branch goes stale?

**Comment by fzowl** [comment]
@bsbodden The failed tests seem to be related to GCP credentials, not during the code tests but during the Authenticate to GCP step, which is in your control, I believe. Can you please help me out with what the issue could be here?
Screenshot 2024-06-08 at 12 14 54

**Comment by tylerhutcherson** [comment]
Will take a look and see if we can get this wrapped up, thanks team!

**Comment by fzowl** [comment]
@tylerhutcherson Thank you!

**Comment by fzowl** [comment]
@tylerhutcherson Any progress on this? Is there anything i can help?

File: redisvl/utils/vectorize/text/voyageai.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+        from redisvl.utils.vectorize import VoyageAITextVectorizer
+
+        vectorizer = VoyageAITextVectorizer(
+            model="voyage-large-2",
+            api_config={"api_key": "your-voyageai-api-key"} # OR set VOYAGE_API_KEY in your env
+        )
+        query_embedding = vectorizer.embed(
+            text="your input query text here",
+            input_type="search_query"
```
Comment (tylerhutcherson) [inline redisvl/utils/vectorize/text/voyageai.py]: https://docs.voyageai.com/docs/embeddings -- input_types are different than cohere's

File: tests/integration/test_rerankers.py [hunk 2]
```diff
@@ -25,11 +40,11 @@ def hfCrossEncoderRerankerWithCustomModel():
 
 
 # Test for basic ranking functionality
-def test_rank_documents_cohere(cohereReranker):
+def test_rank_documents_cohere(reranker):
```
Comment (tylerhutcherson) [inline tests/integration/test_rerankers.py]: I like the fixture updates here! can we make the test method names generic too

**Review by tylerhutcherson — CHANGES_REQUESTED** [review]
Overall it's great -- just a few things I caught when reviewing. I will get an API key today to set up in our CI. Also looks like there is an issue with the lockfile:

https://github.com/redis/redis-vl-python/actions/runs/10885688987/job/30434690376?pr=152

**Comment by tylerhutcherson** [comment]
FYI @fzowl im working on fixing tests over here: https://github.com/redis/redis-vl-python/pull/223
will merge when I have it all working. Needed to address the lockfile, formatting/linting, and one other thing.

**Comment by fzowl** [comment]
@tylerhutcherson Thank you very much! I was struggling a bit with the lockfile, i tried to update it but it was hanging every time for me. Thank you very much for your help!

**Comment by tylerhutcherson** [comment]
> @tylerhutcherson Thank you very much! I was struggling a bit with the lockfile, i tried to update it but it was hanging every time for me. Thank you very much for your help!

No problem! We're happy to support more integrations, and sorry this took so long. We just had a slew of customer requests we needed to focus on first. @fzowl I will close this PR in favor of the other, and plan to release soon. Will VoyageAI do any kind of announcement about the new integration or maybe host an example in your docs? That would be rad.

**Comment by fzowl** [comment]
@tylerhutcherson Yes, i will add RedisVL to our integrations in our site.
I have one more question, is the API documentation automatically generated or shall i extend with VoyageAI? (https://www.redisvl.com/api/vectorizer.html)

**Pull request closed without merging**